File: packages/shared/src/constants/order-status.ts
/**
 * 订单状态 — 镜像后端 OrderStatus 枚举
 */

export const ORDER_STATUS = {
  PENDING: 'pending',
  PAID: 'paid',
  SHIPPED: 'shipped',
  DELIVERED: 'delivered',
  COMPLETED: 'completed',
  CANCELLED: 'cancelled',
  REFUNDED: 'refunded',
} as const;

export type OrderStatus = (typeof ORDER_STATUS)[keyof typeof ORDER_STATUS];

export const ORDER_STATUS_LABEL: Record<OrderStatus, string> = {
  pending: '待付款',
  paid: '已付款',
  shipped: '已发货',
  delivered: '已签收',
  completed: '已完成',
  cancelled: '已取消',
  refunded: '已退款',
};

File: apps/ios/Shop/Feature/Order/Data/Model/OrderStatus.swift
import SwiftUI

enum OrderStatus: String, CaseIterable, Sendable {
    case all = ""
    case pending, paid, shipped, delivered, completed, cancelled, refunded

    var title: String {
        switch self {
        case .all: "All"
        case .pending: "Pending"
        case .paid: "Paid"
        case .shipped: "Shipped"
        case .delivered: "Delivered"
        case .completed: "Completed"
        case .cancelled: "Cancelled"
        case .refunded: "Refunded"
        }
    }

    var icon: String {
        switch self {
        case .all: "list.bullet"
        case .pending: "clock"
        case .paid: "creditcard"
        case .shipped: "shippingbox"
        case .delivered: "house"
        case .completed: "checkmark.circle"
        case .cancelled: "xmark.circle"
        case .refunded: "arrow.uturn.left.circle"
        }
    }

    var color: Color {
        switch self {
        case .all: .shopText
        case .pending: .orange
        case .paid: .shopTeal
        case .shipped: .blue
        case .delivered: .purple
        case .completed: .green
        case .cancelled: .shopTextSecondary
        case .refunded: .red
        }
    }
}
